timemachine/fe/free_energy.py

```diff
@@ -1181,10 +1221,18 @@ def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> Trajectory:
 
             md_params_replica = replace(md_params, n_frames=1, n_eq_steps=0, seed=state_idx + current_frame)
 
-            return sample_with_context(context, md_params_replica, temperature, ligand_idxs, max_buffer_frames=100)
+            assert md_params_replica.n_frames == 1
+            # Get the next set of frames from the iterator, which will be the only value returned
+            frame, box, velos = next(sample_with_context_iter(context, md_params_replica, temperature, ligand_idxs, 1))
+            assert frame.shape[0] == 1
+
+            final_barostat_volume_scale_factor = barostat.get_volume_scale_factor() if barostat is not None else None
+
+            return frame, box, velos, final_barostat_volume_scale_factor
 
-        def replica_from_samples(traj: Trajectory) -> CoordsVelBox:
-            return CoordsVelBox(traj.frames[-1], traj.final_velocities, traj.boxes[-1])
+        def replica_from_samples(last_sample: Tuple[NDArray, NDArray, NDArray, Optional[float]]) -> CoordsVelBox:
+            frame, box, velos, _ = last_sample
+            return CoordsVelBox(frame.squeeze(), velos.squeeze(), box.squeeze())
```

Comment:
nit: preference for explicit slicing instead of `squeeze` here. Is the following equivalent?
```suggestion
            return CoordsVelBox(frame[0], velos[0], box[0])
```
(might be missing something)

Reply:
Could this actually be moved into `sample_replica`? (seems cleaner than having the latter return arrays with size 1 in the first dimension)

Reply:
Moved into `sample_replica` and used explicit slicing in https://github.com/proteneer/timemachine/pull/1334/commits/f359991bb437c04636ae00d82dbad458860be13f

Reply:
Moved into `sample_replica` and used explicit slicing in https://github.com/proteneer/timemachine/pull/1334/commits/f359991bb437c04636ae00d82dbad458860be13f